url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/execution/AgeConverter.java

```diff
@@ -0,0 +1,585 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package uk.ac.leeds.ccg.andyt.generic.execution;
+
+import java.util.Random;
+
+/**
+ * For converting and classifying age groups.
+ */
+public class AgeConverter {
+
+    /**
+     * @param age
+     * @return short of age which is by default 75 and is otherwise:
+     * 0,16,20,25,30,45,60,65,70
+     */
+    public static short getAgeClass1(int age) {
+        if (age < 16) {
+            return 0;
+        }
+        if (age < 20) {
+            return 16;
+        }
+        if (age < 25) {
+            return 20;
+        }
+        if (age < 30) {
+            return 25;
+        }
+        if (age < 45) {
+            return 30;
+        }
+        if (age < 60) {
+            return 45;
+        }
+        if (age < 65) {
+            return 60;
+        }
+        if (age < 70) {
+            return 65;
+        }
+        if (age < 75) {
+            return 70;
+        }
+        return 75;
+    }
+
+    /**
+     * @param age
+     * @return short of age which is by default 60 and is otherwise:
+     * 0,20,30
+     */
+    public static short getAgeClass2(int age) {
+        if (age < 20) {
+            return 0;
+        }
+        if (age < 30) {
+            return 20;
+        }
+        if (age < 60) {
+            return 30;
+        }
+        return 60;
+    }
+
+    /**
+     * @param age
+     * @return short of age which is by default 75 and is otherwise:
+     * 0-15,16,20,25,30,45,60,65,70,75,80,85,90
+     */
+    public static short getAgeClass3(int age) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.